av_metrics/src/video/mod.rs

```diff
@@ -211,8 +171,8 @@ trait VideoMetric {
         let mut frame_no = 0;
         while frame_limit.map(|limit| limit > frame_no).unwrap_or(true) {
             if decoder1.get_bit_depth() > 8 {
-                let frame1 = decoder1.read_video_frame::<u16>();
-                let frame2 = decoder2.read_video_frame::<u16>();
+                let frame1 = decoder1.read_video_frame::<u16>(&decoder1.get_video_details());
```

Comment:
Probably we could get the video details for each video once outside of the loop, and reuse it. Same for Lines 184-185.

Reply:
Indeed, updated